packages/protocol/lib/backward/ast-code.ts

```diff
@@ -0,0 +1,380 @@
+// tslint:disable: max-classes-per-file
```

Comment:
the second one is too long IMO. And I don't have any strong opinions about either `backward` or `compatibility`. Since one is already implemented I'd leave that, unless there is any strong argument for the other. What do you think?

Reply:
Without context, I don't think I would have much of an idea of what I'm looking at when I see `lib/backward`, whereas `lib/compatibility` is much clearer in the library's intent.

Reply:
renamed to compatibility